### PR TITLE
Okta integration

### PR DESCRIPTION
**Pull request by aviadl**
fixes https://github.com/demisto/server/issues/4194
@liorkol As we talked please review and add scripts.
@Spivaka please review as well, but DO NOT merge
We can do this together if you want

### CODE REVIEW

File: Integrations/integration-Okta.yml [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+        'employeeNumber',
+        'costCenter',
+        'organization',
+        'division',
+        'department',
+        'managerId',
+        'manager'
+    ];
+
+    var fixUrl = function(base) {
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Didn't we say we move this to the _script_integration_template?

File: Integrations/integration-Okta.yml [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+        res = base;
+        if (base && base[base.length - 1] != '/') {
+            res = res + '/';
+        }
+        return res;
+    };
+
+    var parseResponse = function(resp) {
+        if (resp.StatusCode === 200) {
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Maybe should also be in top level and get the parsing func (JSON.parse in this case)

File: Integrations/integration-Okta.yml [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+        url = url + uri;
+         res = http(
+            url,
+            {
+                Method: 'GET',
+                Headers: {
+                    Authorization: ['SSWS ' + params.apitoken]
+                }
+            },
+            true
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: It's always insecure? shouldn't be a param?

File: Integrations/integration-Okta.yml [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+        var keys = Object.keys(args);
+        for (var i = 0 ; i < profileArgs.length; i++) {
+            if (args[profileArgs[i]]) {
+                profile[profileArgs[i]] = args[profileArgs[i]];
+            }
+        }
+        return profile;
+    };
+
+    /*
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Remove commented out code

File: Integrations/integration-Okta.yml [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+      if (args.password) {
+          var  pswd = { value: args.password };
+          creds.password = pswd;
+      }
+      if (args.passwordQuestion || args.passwordAnswer) {
+          var recovery = {
+              question: args.passwordQuestion,
+              answer: args.passwordAnswer
+          };
+          creds.recovery_question = recovery;
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Why not in a single line? 
creds.recovery_question = {
             question: args.passwordQuestion,
             answer: args.passwordAnswer
         };

File: Integrations/integration-Okta.yml [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+      }
+      if (args.providerName || args.providerType) {
+          var provider = {
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: same here...

File: Integrations/integration-Okta.yml [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+                return 'ok';
+            } else if (resp.Status) {
+                return resp.Status;
+            } else {
+                return resp;
+            }
+            break;
+        case 'okta-deactivate-user':
+            var userId = getUserId(args, params);
+            var res = post(args, params, 'api/v1/users/' + userId + '/lifecycle/deactivate');
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Maybe it's a matter of flavour, but I prefer 
return post(args, params, 'api/v1/users/' + getUserId(args, params) + '/lifecycle/deactivate')

File: Integrations/integration-Okta.yml [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+            profile.login = args.username;
+            var body = {
+                profile: profile,
+                credentials: cred
+            };
+            var uri = '/api/v1/users/' + userId;
+            var res = post(args, params, uri, body);
+            return res;
+        default:
+            // You can use args[argName] or args.argName to get a specific arg. args are strings.
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Remove default and comments...

File: Integrations/integration-Okta.yml [hunk 9]
```diff
@@ -0,0 +1,509 @@
...
+            },
+            true
+        );
+        return res;
+    };
+
+    var post = function(args, params, uri, body) {
+        url = fixUrl(params.url);
+        url = url + uri;
+        readyBody = null;
```
Comment (Spivaka) [inline Integrations/integration-Okta.yml]: Why not:
readyBody = body ? JSON.stringify(body) : undefined;
Too much go? :)

**Comment by aviadl** [comment]
@Spivaka @liorkol fixed the comments which i thought were relevant
@liorkol wating for you to review the texts, and add scripts

**Comment by liorkol** [comment]
#fixes https://github.com/demisto/server/issues/4263

**Pull request merged**